File: src/java/eu/sig/training/ch07/ApplicationLauncher.java
package eu.sig.training.ch07;

@SuppressWarnings("unused")
// tag::ApplicationLauncher[]
public class ApplicationLauncher {

    public static void main(String[] args) {
        CloudServerFactory factory;
        if (args[1].equals("-azure")) {
            factory = new AzureCloudServerFactory();
        } else {
            factory = new AWSCloudServerFactory();
        }
        CloudServer computeServer = factory.launchComputeServer();
        CloudServer databaseServer = factory.launchDatabaseServer();
        // end::ApplicationLauncher[]
    }

}

File: src/csharp/eu/sig/training/ch07/ApplicationLauncher.cs
namespace eu.sig.training.ch07
{
    // tag::ApplicationLauncher[]
    public class ApplicationLauncher
    {

        public static void Main(string[] args)
        {
            ICloudServerFactory factory;
            if (args[1].Equals("-azure"))
            {
                factory = new AzureCloudServerFactory();
            }
            else
            {
                factory = new AWSCloudServerFactory();
            }
            ICloudServer computeServer = factory.LaunchComputeServer();
            ICloudServer databaseServer = factory.LaunchDatabaseServer();
            // end::ApplicationLauncher[]
        }
    }
}
